service/resolver/cri.go

```diff
@@ -136,18 +137,20 @@ func RegistryHostsFromCRIConfig(ctx context.Context, config Registry, credsFuncs
 			}
 
 			var (
-				transport = http.DefaultTransport.(*http.Transport).Clone()
-				client    = &http.Client{Transport: transport}
-				config    = config.Configs[u.Host]
+				rclient = rhttp.NewClient()
+				config  = config.Configs[u.Host]
 			)
 
 			if config.TLS != nil {
-				transport.TLSClientConfig, err = getTLSConfig(*config.TLS)
-				if err != nil {
-					return nil, errors.Wrapf(err, "get TLSConfig for registry %q", e)
+				if tr, ok := rclient.HTTPClient.Transport.(*http.Transport); ok {
```

Comment:
Should return an error if the tls config cannot be applied (i.e. `!ok`).

Reply:
Thanks, I'll update it.

Reply:
Fixed.